src/main/java/com/getindata/connectors/http/internal/retry/HttpClientWithRetry.java

```diff
@@ -0,0 +1,88 @@
+package com.getindata.connectors.http.internal.retry;
+
+import com.getindata.connectors.http.HttpStatusCodeValidationFailedException;
+import com.getindata.connectors.http.internal.status.HttpResponseChecker;
+import io.github.resilience4j.retry.Retry;
+import io.github.resilience4j.retry.RetryConfig;
+import io.github.resilience4j.retry.RetryRegistry;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.metrics.MetricGroup;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.function.Supplier;
+
+@Slf4j
+public class HttpClientWithRetry {
+
+    private final HttpClient httpClient;
+    @Getter
+    private final HttpResponseChecker responseChecker;
+    private final Retry retry;
+
+    @Builder
+    HttpClientWithRetry(HttpClient httpClient,
```

Comment:
I am wondering what happens with OIDC, the short lived bearer token may need to be regenerated if the retries occur after the token has expired). Is this regeneration check done for the retries?